src/main/java/net/dries007/tfc/objects/blocks/plants/BlockCreepingPlantTFC.java

```diff
@@ -0,0 +1,279 @@
+            TABLE.put(plant, new EnumMap<>(Plant.PlantType.class));
+        TABLE.get(plant).put(type, this);
+
+        this.plant = plant;
+        this.type = type;
+        this.setDefaultState(this.blockState.getBaseState().withProperty(GROWTHSTAGE, CalenderTFC.getMonthOfYear().id()).withProperty(DOWN, Boolean.valueOf(false)).withProperty(UP, Boolean.valueOf(false)).withProperty(NORTH, Boolean.valueOf(false)).withProperty(EAST, Boolean.valueOf(false)).withProperty(SOUTH, Boolean.valueOf(false)).withProperty(WEST, Boolean.valueOf(false)));
+    }
+
+    public boolean canConnectTo(IBlockAccess worldIn, BlockPos pos, EnumFacing facing)
+    {
+        IBlockState iblockstate = worldIn.getBlockState(pos);
+        BlockFaceShape blockfaceshape = iblockstate.getBlockFaceShape(worldIn, pos, facing);
+        Block block = iblockstate.getBlock();
+        return blockfaceshape == BlockFaceShape.SOLID || block instanceof BlockFence;
+    }
+
+    @Override
+    public IBlockState getActualState(IBlockState state, IBlockAccess worldIn, BlockPos pos)
+    {
+        return state.withProperty(TIME, state.getValue(TIME))
+            .withProperty(GROWTHSTAGE, CalenderTFC.getMonthOfYear().id())
+            .withProperty(DOWN, canPlantConnectTo(worldIn, pos, EnumFacing.DOWN))
+            .withProperty(UP, canPlantConnectTo(worldIn, pos, EnumFacing.UP))
+            .withProperty(NORTH, canPlantConnectTo(worldIn, pos, EnumFacing.NORTH))
+            .withProperty(EAST, canPlantConnectTo(worldIn, pos, EnumFacing.EAST))
+            .withProperty(SOUTH, canPlantConnectTo(worldIn, pos, EnumFacing.SOUTH))
+            .withProperty(WEST, canPlantConnectTo(worldIn, pos, EnumFacing.WEST));
+    }
+
+    @Override
+    public int getMetaFromState(IBlockState state)
+    {
+        return 0;
+    }
+
+    @Override
+    protected BlockStateContainer createBlockState()
+    {
+        return new BlockStateContainer(this, new IProperty[] {DOWN, UP, NORTH, EAST, WEST, SOUTH, GROWTHSTAGE, TIME});
+    }
+
+    @Override
+    public Block.EnumOffsetType getOffsetType()
+    {
+        return EnumOffsetType.NONE;
+    }
+
+    @Override
+    protected boolean canSustainBush(IBlockState state)
+    {
+        return true;
+    }
+
+    @Override
+    public boolean canBlockStay(World worldIn, BlockPos pos, IBlockState state)
+    {
+        for (EnumFacing face : EnumFacing.values())
+        {
+            IBlockState blockState = worldIn.getBlockState(pos.offset(face));
+            if (blockState.getBlockFaceShape(worldIn, pos.offset(face), face.getOpposite()) == BlockFaceShape.SOLID || blockState.getBlock() instanceof BlockFence)
+            {
+                return plant.isValidLocation(ClimateTFC.getHeightAdjustedBiomeTemp(worldIn, pos), ChunkDataTFC.getRainfall(worldIn, pos));
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, BlockPos pos)
```

Comment:
See [Metal Sheet](https://github.com/alcatrazEscapee/TerraFirmaCraft/blob/2be3b2ccb61a1c36d9febc401262b526f913a832/src/main/java/net/dries007/tfc/objects/blocks/metal/BlockMetalSheet.java#L158) for a nicer way to handle this